### ISSUE TITLE
get_first false negative: VecDeque.get(0)

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#11695]
### Summary

The rationale for preferring `vec.first()` over `vec.get(0)` also applies to preferring `vecdeque.front()` over `vecdeque.get(0)`.

### Lint Name

get_first

### Reproducer

```rust
use std::collections::VecDeque;

fn main() {
    let vec = Vec::from(['a', 'b', 'c']);
    println!("{:?}", vec.get(0));

    let vecdeque = VecDeque::from(['a', 'b', 'c']);
    println!("{:?}", vecdeque.get(0));
}
```

Today, `cargo clippy` only suggests replacing `vec.get(0)` with `vec.first()`. It does not suggest replacing `vecdeque.get(0)` with `vecdeque.front()`.

### Version

```text
rustc 1.75.0-nightly (1c05d50c8 2023-10-21)
binary: rustc
commit-hash: 1c05d50c8403c56d9a8b6fb871f15aaa26fb5d07
commit-date: 2023-10-21
host: aarch64-unknown-linux-gnu
release: 1.75.0-nightly
LLVM version: 17.0.3
```

**Comment by matthri** [comment]
Hi, may I try to fix this?

**Comment by y21** [comment]
@matthri sure! You can write `@rustbot claim` to assign yourself this issue. The relevant code is in `clippy_lints/src/methods/get_first.rs`
